nixos/modules/services/hardware/handheld-daemon.nix

```
@@ -0,0 +1,43 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+with lib; let
+  cfg = config.services.handheldDaemon;
+in
+{
+  options.services.handheldDaemon = {
+    enable = mkEnableOption "Enable Handheld Daemon";
+
+    user = mkOption {
+      type = types.str;
+      description = lib.mdDoc ''
+        The user to run Handheld Daemon with.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ pkgs.handheld-daemon ];
+    services.udev.packages = [ pkgs.handheld-daemon ];
+    systemd.packages = [ pkgs.handheld-daemon ];
+
+    systemd.services.handheldDaemon = {
+      description = "Handheld Daemon";
+
+      wantedBy = [ "multi-user.target" ];
+
+      restartIfChanged = true;
+
+      serviceConfig = {
+        ExecStart = "${ pkgs.handheld-daemon }/bin/hhd --user ${ cfg.user }";
```

Maybe a package option could be added?

```suggestion
    enable = mkEnableOption "Enable Handheld Daemon";
    
    package = mkPackageOption pkgs "handheld-daemon" { };

    user = mkOption {
      type = types.str;
      description = lib.mdDoc ''
        The user to run Handheld Daemon with.
      '';
    };
  };

  config = mkIf cfg.enable {
    environment.systemPackages = [ pkgs.handheld-daemon ];
    services.udev.packages = [ pkgs.handheld-daemon ];
    systemd.packages = [ pkgs.handheld-daemon ];

    systemd.services.handheldDaemon = {
      description = "Handheld Daemon";

      wantedBy = [ "multi-user.target" ];

      restartIfChanged = true;

      serviceConfig = {
        ExecStart = "${ cfg.package }/bin/hhd --user ${ cfg.user }";
```

---

What would that do?

Are you trying to make the rule reusable with other packages?

(This is my first nixpkg.)

---

It would make it easier to override the package, in case you'd want to do any changes to the package. Example:

```nix
services.handheldDaemon.package = pkgs.handheld-daemon.override {
  python3 = pkgs.python38;
};
```

---

Interesting.  Wouldn't have expected someone would want to do that.

I'm guessing you'd also need to use `cfg.package` in these places?

```nix
    environment.systemPackages = [ pkgs.handheld-daemon ];
    services.udev.packages = [ pkgs.handheld-daemon ];
    systemd.packages = [ pkgs.handheld-daemon ];
```

---

> I'm guessing you'd also need to use `cfg.package` in these places?

Correct